pkg/prometheus/promcfg.go

```diff
@@ -2318,11 +2320,23 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 		cfg = append(cfg, yaml.MapItem{Key: "scrape_timeout", Value: *sc.Spec.ScrapeTimeout})
 	}
 
-	if sc.Spec.RelabelConfigs != nil {
+	if len(sc.Spec.RelabelConfigs) > 0 {
 		relabelings = append(relabelings, generateRelabelConfig(labeler.GetRelabelingConfigs(sc.TypeMeta, sc.ObjectMeta, sc.Spec.RelabelConfigs))...)
-		cfg = append(cfg, yaml.MapItem{Key: "relabel_configs", Value: relabelings})
 	}
 
+	alreadyHasShardLabel := false
+	for _, relabelConfig := range sc.Spec.RelabelConfigs {
+		if relabelConfig.Action == "hashmod" {
+			level.Debug(cg.logger).Log("msg", "found existing hashmod relabeling rule, skipping", "relabeling", relabelConfig)
```

Comment:
(nit) RelabelConfig doesn't implement the Stringer interface so it won't log cleanly.
```suggestion
			level.Debug(cg.logger).Log("msg", "found existing hashmod relabeling rule, skipping", "idx", i)
```